pyfstat/core.py

```diff
@@ -1565,6 +1565,28 @@ def __init__(
         self.init_computefstatistic()
         self.init_semicoherent_parameters()
 
+    def _init_semicoherent_window_trick(self):
+        """
+        Use this window to compute the semicoherent Fstat using TransientFstatMaps.
+        This way we are able to decouple this clever shenanigan from the actual
+        usage of a transient window.
+        """
+        self._semicoherent_window_trick = lalpulsar.transientWindowRange_t()
```

Comment:
Is it "pythonic" to have variables (as opposed to methods) with `_`? I don't think we're doing that anywhere else, at least?

Reply:
Another, semantically more correct and possibly cleaner option, would be to rename this one `self.semicoherentWindowRange` and the existing `self.windowRange` into `self.transientWindowRange`.

Reply:
> Is it "pythonic" to have variables (as opposed to methods) with `_`? I don't think we're doing that anywhere else, at least?

[Somehow, yes](https://www.python.org/dev/peps/pep-0008/). The intention is to say "This variable is spookier than you think; don't mess around with it unless you know what you are doing".



> Another, semantically more correct and possibly cleaner option, would be to rename this one `self.semicoherentWindowRange` and the existing `self.windowRange` into `self.transientWindowRange`.

I'd say `self.windowRange` was not doing what it  looks like it's doing... mainly because it was only used to do the Transient trick and nothing else.

I'd prefer to hide this variable since it's not a transient window to compute semicoherent transient statistics, but a transient window to do some "look-at-me-I'm-using-polymorphism" magic

Reply:
Since we have a lot of other clearly "internal"/"spooky" variables that aren't name that way, I think I'd prefer to not introduce this convention just for this one. Calling it `self.semicoherentWindowRange` should make clear what it does.

Reply:
Ok, I like the internal consistency argument.